### PR TITLE
Djulian/tdd

### PR DESCRIPTION
**Pull request by juleahn**
Added TestCoursePreq.py for TDD

### CODE REVIEW

File: backend/integration_test.py [hunk 1]
```diff
@@ -6,13 +6,17 @@
 suite = unittest.TestSuite()
 
 suite.addTests(
-    loader.loadTestsFromTestCase(TestTrainerController.TestTrainerController)
+    [
+        loader.loadTestsFromTestCase(TestCoursePreqModel.TestCoursePreq),
+        loader.loadTestsFromTestCase(TestTrainerController.TestTrainerController),
```
Comment (alphatrl) [inline backend/integration_test.py:11]: Fix formating

File: backend/integration_test.py [hunk 2]
```diff
@@ -6,13 +6,17 @@
...
+    ]
 )
 
 # initialize a runner, pass it your suite and run it
 runner = unittest.TextTestRunner(verbosity=3)
 result = runner.run(suite)
 
+
```
Comment (alphatrl) [inline backend/integration_test.py:19]: remove extra line

File: backend/main.py [hunk 3]
```diff
@@ -334,6 +334,22 @@ def edit_class():
         print(e, flush=True)
         return error.throw_error(type="create_class", message=str(e), status_code=400)
 
+@app.route("/api/learners")
```
Comment (alphatrl) [inline backend/main.py:337]: delete added function

File: backend/integration_test.py (hunk 2 above)
Comment (alphatrl) [inline backend/integration_test.py:19]: remove extra line

File: backend/main.py (hunk 3 above)
Comment (alphatrl) [inline backend/main.py:337]: remove old function.

File: backend/integration_test.py (hunk 2 above)
Comment (alphatrl) [inline backend/integration_test.py:19]: Remove extra line

File: backend/main.py (hunk 3 above)
Comment (alphatrl) [inline backend/main.py:337]: remove function. it has already been removed in main branch

File: backend/integration_test.py (hunk 2 above)
Comment (alphatrl) [inline backend/integration_test.py:19]: remove extra line

File: backend/main.py [hunk 4]
```diff
@@ -334,6 +334,22 @@ def edit_class():
...
+def learners():
+    search_first_name = request.args.get('first_name')
+    if search_first_name:
+        learner_list = Learner.Learner.query.filter(Learner.Learner.name.contains(search_first_name))
+    else:
```
Comment (alphatrl) [inline backend/main.py:342]: remove function. already removed from main branch

File: backend/tests/TestApp.py [hunk 5]
```diff
@@ -193,4 +193,4 @@ def init_db(self):
 
 
 if __name__ == "__main__":
-    unittest.main()
+    unittest.main()
```
Comment (alphatrl) [inline backend/tests/TestApp.py:196]: add extra line below

File: backend/tests/TestTrainerController.py [hunk 6]
```diff
@@ -106,4 +106,4 @@ def test_is_trainer_false(self):
```
Comment (alphatrl) [inline backend/tests/TestTrainerController.py:106]: add extra line below. thanks

File: backend/tests/__init__.py [hunk 7]
```diff
@@ -1,4 +1,5 @@
 __all__ = [
```
Comment (alphatrl) [inline backend/tests/__init__.py:1]: add extra line below

File: webapp/src/router/index.js [hunk 8]
```diff
@@ -96,6 +96,11 @@ const routes = [
     name: 'Logout',
     component: () => import('../views/Auth/Logout.vue'),
   },
+  {
+    path: '/learners',
```
Comment (alphatrl) [inline webapp/src/router/index.js:100]: remove this

File: webapp/src/views/assignLearners.vue [hunk 9]
```diff
@@ -0,0 +1,17 @@
+<template>
```
Comment (alphatrl) [inline webapp/src/views/assignLearners.vue:1]: remove

Comment (alphatrl) [reply webapp/src/views/assignLearners.vue:1]: delete file

File: webapp/yarn.lock [hunk 10]
```diff
@@ -1513,6 +1513,13 @@
   dependencies:
     eslint-config-prettier "^6.0.0"
 
+"@vue/eslint-config-prettier@^6.0.0":
```
Comment (alphatrl) [inline webapp/yarn.lock:1516]: remove this change

**Review by alphatrl — CHANGES_REQUESTED** [review]
remove all changes made due to merge from the old branch

**Comment by alphatrl** [comment]
Remove database.db  in /backend and /backend/tests

File: backend/integration_test.py (hunk 2 above)
Comment (juleahn) [reply backend/integration_test.py:19]: Deleted line

File: backend/main.py (hunk 3 above)
Comment (juleahn) [reply backend/main.py:337]: removed function

File: backend/integration_test.py (hunk 2 above)
Comment (juleahn) [reply backend/integration_test.py:19]: removed line

File: backend/main.py (hunk 3 above)
Comment (juleahn) [reply backend/main.py:337]: removed old function

**Review by alphatrl — APPROVED** [review]

**Pull request merged**